pkg/p2p/peer/processing/chainsync/sync.go

```diff
@@ -99,7 +99,7 @@ func (s *ChainSynchronizer) Synchronize(blkBuf *bytes.Buffer, peerInfo string) e
 			return err
 		}
```

Comment:
Is reader from `r := bufio.NewReader(blkBuf)` redundant? 

```go

// Write bufio.Reader into a bytes.Buffer and unmarshal it so we can send it over the event bus.
		buf := new(bytes.Buffer)
		if _, err := buf.ReadFrom(r); err != nil {
			return err
		}

		if err := message.UnmarshalBlock(buf, blk); err != nil {
			return err
		}
```

Instead, we can use ` message.UnmarshalBlock(blkBuf, blk);` or I'm missing something.

Reply:
It's not - we use it to peek the block height at the start of the function. Helps us determine whether we should start syncing, forward the block, or discard it.

Reply:
That's correct. Forgot this. Thanks!